p2p/src/lib.rs

```diff
@@ -87,9 +133,9 @@ where
         Ok(Self {
             network: NetworkingBackend::new(addr, &[], &[]).await?,
             config,
-            peer_cnt: AtomicU64::default(),
             peer_backlock,
             peers: HashMap::new(),
```

Comment:
Since we know how many peers there can be, we have a max number after all, it might be worth using `with_capacity()` with the peers hashmap so we can guarantee avoid the pain of reallocating memory as we add elements to it.